java/spring/containerWithTests/Application/sampleWebApp.FunctionalTests/src/main/java/functional_tests/SampleFunctionalTest.java

```diff
@@ -32,8 +33,11 @@ public void testAssertTitleWithSelenium() throws InterruptedException, Assertion
             System.out.println(isAlertPresent());
             driver.switchTo().alert().accept();
         }
-        int numRetries = 5;
-        for (int i = 0; i < numRetries; i++)
+
+        long currentTimestamp = (new Date()).getTime();
+        long endTimestamp = currentTimestamp + 60*10*1000;
+
+        while(currentTimestamp <= endTimestamp + 10000)

```

Comment:
no need to have a condition here. You an have while(1). Instead use the condition in catch block

Reply:
I'm not sure what you meant, If we use the condition in catch block to break out of the loop, in the case where webapp is successfully deployed, the loop will run forever.